packages/amplify-codegen-e2e-core/src/init/cra.ts

```diff
@@ -0,0 +1,36 @@
+/* commands for a Create React App */
+import { nspawn as spawn } from '..';
+
+const defaultSettings = {
+  disableCIDetection: false,
+};
+
+export function craInstall(cwd: string, settings: Object = {}): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const s = { ...defaultSettings, ...settings };
+    const chain = spawn('npm', ['install'], { cwd, stripColors: true, disableCIDetection: s.disableCIDetection });
+
+    chain.run((err: Error) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve();
+      }
+    });
```

Comment:
`runAsync` is a nice way to reduce the if/else here. This block could be rewritten.

```typescript
return spawn('npm', ['install'], { cwd, stripColors: true, disableCIDetection: s.disableCIDetection }).runAsync();
```

Reply:
And same for the following fn as well.

Reply:
Doesn't look like we have a `runAsync` https://github.com/aws-amplify/amplify-codegen/blob/main/packages/amplify-codegen-e2e-core/src/utils/nexpect.ts#L56-L72.